### PR TITLE
Nfft

### PR DESCRIPTION
**Pull request by LElgueddari**
Making the NFFT compatible with the 3D and more general. Adding some documentation and changing the general utils tools.

Still waiting for the NFFT to be integrated with Travis so we can run the tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ before_install:
         export CPLUS_INCLUDE_PATH=/usr/include/python3.4m;
       fi
     - if [ $TRAVIS_PYTHON_VERSION == "3.5" ]; then
-        export CPLUS_INCLUDE_PATH=/opt/python/3.5.5/include/python3.5m;
+        export CPLUS_INCLUDE_PATH=/opt/python/3.5.6/include/python3.5m;
```
Comment (LElgueddari) [inline .travis.yml]: New python version is used in Travis

File: pysap/plugins/mri/reconstruct/fourier.py [hunk 2]
```diff
@@ -152,20 +174,20 @@ def __init__(self, samples, shape):
...
     def op(self, img):
         """ This method calculates the masked non-cartesian Fourier transform
-        of a 2-D image.
+        of a N-D data.
 
         Parameters
         ----------
         img: np.ndarray
-            input 2D array with the same shape as the mask.
+            input MD array with the same shape as the mask.
```
Comment (sfarrens) [inline pysap/plugins/mri/reconstruct/fourier.py]: It's not really important, but did you mean to put "MD" or should it be "ND"?

File: pysap/plugins/mri/reconstruct/fourier.py [hunk 3]
```diff
@@ -120,27 +120,49 @@ def adj_op(self, x):
         return pfft.ifft2(self._mask * x)
 
 
-class NFFT2(FourierBase):
-    """ Standard 2D non catesian Fast Fourrier Transform class
+class NFFT(FourierBase):
+    """ ND non catesian Fast Fourrier Transform class
+    The NFFT will normalize in a symmetric way the direct and adjoint operator.
```
Comment (sfarrens) [inline pysap/plugins/mri/reconstruct/fourier.py]: I think the specific type of normalisation used should be described in this docstring somwhere.

Comment (LElgueddari) [reply pysap/plugins/mri/reconstruct/fourier.py]: the real problem s that the FFT and the NFFT doesn't take the same convention and this is weird for the same package. so we do have the choice, keep the unitary convention which will define the FFT as a orthogonal bases or keep the original convention of the FFT

Comment (sfarrens) [reply pysap/plugins/mri/reconstruct/fourier.py]: Do you have any preference for which convention should be used?

**Comment by LElgueddari** [comment]
@sfarrens  and @AGrigis: I will wait PR #58 yo be merged, then I will uncomment fourier unit test

File: .travis.yml (hunk 1 above)
Comment (sfarrens) [reply .travis.yml]: PR #58 has been merged, please pull latest commits.

File: sparse2d/python/CMakeLists.txt [hunk 4]
```diff
@@ -58,11 +58,19 @@ project(pysparse)
 
   # Build and link the pylib module
   add_library(pysparse SHARED pysparse.cpp)
-  target_link_libraries(pysparse ${OpenMP_CXX_LIBRARIES} ${sparse2d_LIBRARIES} ${cfitsio_LIBRARIES} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES})
+  if(APPLE)
+    target_link_libraries(pysparse ${OpenMP_CXX_LIBRARIES} ${sparse2d_LIBRARIES} ${cfitsio_LIBRARIES} ${Boost_LIBRARIES})
+  else(APPLE)
+    target_link_libraries(pysparse ${OpenMP_CXX_LIBRARIES} ${sparse2d_LIBRARIES} ${cfitsio_LIBRARIES} ${Boost_LIBRARIES} ${PYTHON_LIBRARIES})
+  endif(APPLE)
```
Comment (sfarrens) [inline sparse2d/python/CMakeLists.txt]: Changes included in PR #57, please pull latest commits

File: .travis.yml (hunk 1 above)
Comment (LElgueddari) [reply .travis.yml]: Done

File: sparse2d/python/CMakeLists.txt (hunk 4 above)
Comment (LElgueddari) [reply sparse2d/python/CMakeLists.txt]: Done

**Pull request merged**